controllers/drplacementcontrol_controller.go

```diff
@@ -976,25 +1046,71 @@ func (d *DRPCInstance) runInitialDeployment() (bool, error) {
 		homeClusterNamespace = d.drpcPlacementRule.Status.Decisions[0].ClusterNamespace
 	}
 
-	if homeCluster == "" {
-		err := fmt.Errorf("PreferredCluster not set and unable to find home cluster in DRPCPlacementRule (%v)",
-			d.drpcPlacementRule)
-		// needStatusUpdate is not set. Still better to capture the event to report later
-		rmnutil.ReportIfNotPresent(d.reconciler.eventRecorder, d.instance, corev1.EventTypeWarning,
-			rmnutil.EventReasonDeployFail, err.Error())
+	return homeCluster, homeClusterNamespace
+}
 
-		return !done, err
+// isDeployed check to see if the initial deployment is already complete to this
+// homeCluster or elsewhere
+func (d *DRPCInstance) isDeployed(homeCluster string) (string, bool) {
+	if d.isAlreadyDeployedAndProtected(homeCluster) {
```

Comment:
Given that VRG's ClusterDataProtected condition should only prevent DRPC relocate DRAction,  I'm not sure if this check for cluster data protected status here is correct here?

Reply:
You are right.  We decided to not wait for ClusterDataProction in `InitialDeploy` or `Failover`.